scripts/oslextractmeta.py

```diff
@@ -0,0 +1,453 @@
+        compShader['ui']      = shaderUI
+        return compShader
+
+
+#----------------------------------------------------------
+# Functions for handling the shader dictionary
+#----------------------------------------------------------
+
+def getNumberOfShaders(jsonFile):
+    return len(jsonFile['shaders'])
+
+
+def cleanJsonShaders(jsonDict):
+    num_del = 0
+    for shaderpath in jsonDict.keys():
+        if not os.path.isfile(shaderpath):
+            del jsonDict[shaderpath]
+            num_del += 1
+    return (num_del, jsonDict)
+
+
+def existsJsonShader(jsonFile, shaderName):
+    for shader in jsonFile['shaders']:
+        if shader['name'] == shaderName:
+            return True
+        else:
+            return False
+
+
+def writeJsonHeader(filename, numElements):
+    headerDict = dict()
+    headerDict['creator']       = getpass.getuser()
+    headerDict['creation date'] = str(datetime.datetime.now())
+    headerDict['name']          = os.path.basename(filename)
+    headerDict['elements']      = numElements
+    headerDict['last update']   = str(datetime.datetime.now())
+    return headerDict
+
+
+def updateJsonHeader(jsonFile, numElements):
+    headerDict = jsonFile
+    headerDict['last update'] = str(datetime.datetime.now())
+    headerDict['elements'] = numElements
+    return headerDict
+
+
+def cli():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description = '''
+        oslextractmetadata stores the user interface and metadata of a 
+        compiled OSL (openshadinglanguage) shader(s) into a JSON file.
+        The JSON dictionary consists of a 'header' and a 'shader' part.
+        jsondict['shader'] will return a dictionary with all shaders. The
+        user interface of the shader is stored as a sub-dictionary, the
+        metadata can be retrieved using the 'ui' key on the elements, e.g.:
+
+        for x in jsondict['shaders'].values():
+            print x['ui']
+        ''')
+    parser.add_argument('-i', '--input', nargs='+', action='store', dest='files', metavar='compiled shaders',help='List of file(s) to parse.')
+    parser.add_argument('-v', '--verbose', action='store_true', dest='verbosity', help='Increase output verbosity.')
+    parser.add_argument('-o', '--output', nargs=1, action='store', dest='output', required=True, metavar='output file', help="Store shader UI in file.")
+    parser.add_argument('-f', '--file', nargs='+', action='store', dest='read_file', metavar='file',  help="Read file paths from file(s).")
+    parser.add_argument('-U', '--update', action='store_true', dest='update', help="Update existing shader file.")
+    parser.add_argument('-O', '--overwrite', action='store_true', dest='overwrite', help="Overwrite existing files.")
+    parser.add_argument('-c', '--clean', action='store_true', dest='clean', help="Clean file, remove non existant shaders.")
```

Comment:
I'd remove the update option and always overwrite the json file.
Running the script shouldn't be too expensive and it's not something users will do frequently.
Overwriting all the time should be ok and all the code related to update and clean could be removed.